src/test/java/com/nvs/store/service/ProductServiceTest.java

```diff
@@ -0,0 +1,76 @@
+package com.nvs.store.service;
+
+import com.nvs.store.models.product.Product;
+import com.nvs.store.repository.ProductRepository;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import java.math.BigDecimal;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
+public class ProductServiceTest {
+
+    @Autowired
+    private ProductService productService;
+
+    @MockBean
+    private ProductRepository productRepository;
+
+
+    @Test
+    public void getAllProductsTest() {
+        when(productRepository.findAll()).thenReturn(Stream.of(
+                new Product(1L, "strawberry", 25, BigDecimal.valueOf(45.60)),
+                new Product(1L, "banana", 31, BigDecimal.valueOf(37.90))).collect(Collectors.toList()));
+        assertEquals(2, productService.getAllProducts().size());
+    }
+
+    @Test
+    public void getProductTest() {
+        Long id = 1L;
+        Optional<Product> product = Optional.of(new Product(1L, "strawberry", 25, BigDecimal.valueOf(45.60)));
+        Mockito.when(productRepository.findById(id)).thenReturn(product);
+        Optional<Product> productById = productService.getProduct(id);
+        assertEquals(product, productById);
+    }
+
+    @Test
+    public void addProductTest() {
+        Optional<Product> product = Optional.of(new Product(1L, "orange", 49, BigDecimal.valueOf(23.30)));
+        when(productRepository.save(product.get())).thenReturn(product.get());
+        Mockito.when(productRepository.findById(product.get().getId())).thenReturn(product);
+        Optional<Product> productById = productService.getProduct(product.get().getId());
+        assertEquals(product,productById);
+    }
+
+    @Test
+    public void updateProductTest() {
```

Comment:
The scope of unit tests is to check the code added by developer, is not required to test implementation from dependency library.
Current unit test can be simplified by using Mockito#verify and arg check.
First step have to mock productRpository Mockito.when(productRepository.getProductById(id)).thenReturn(product), the service implementation doesn't use "getById" method, this step is to avoid null pointer exception. 
Second step call productService. updateProduct(productId, Product).
Third step to verify that save method of product repository has been called with expected parameters
Mockito.verify(productRepository)
	.save(
		Mockito.argThat(
			new ArgumentMatcher<Product>() {
				@Override
				public boolean matches(Product argument) {
					// match the fields using convenient API
					return false; // or return if expected or not
				}
			}));
}

Reply:
done